### PR TITLE
Initial draft of a lib3h 'integration'

### PR DESCRIPTION
**Pull request by ddd-mtl**
Initial integration with Lib3h by using a Lib3h stub ran in a Lib3hWorker ([lib3h stub PR](https://github.com/holochain/lib3h/pull/21)).
The Lib3h stub is the `RealEngine` struct in Lib3h repo.
Lib3hWorker manages a `RealEngine` instance.
It can also convert a RealEngineConfig from JSON.

Added `Protocol::Lib3h`. End-goal is to have that replace `Protocol::Json` everywhere. For now we have cohabitation.
Added `P2pBackendKind::LIB3H`.
Added LIB3H test mode and corresponding testing functions in `test_bin`.
Added `create_lib3h_config()` for generating a compatible P2pConfig for LIB3H.

### CODE REVIEW

File: net/Cargo.toml [hunk 1]
```diff
@@ -10,6 +10,8 @@ tempfile = "=3.0.7"
 [dependencies]
 directories = "=1.0.2"
 failure = "=0.1.5"
+holochain_lib3h = { path = "../../lib3h/lib3h" }
+holochain_lib3h_protocol = { path = "../../lib3h/lib3h_protocol" }
```
Comment (neonphog) [inline net/Cargo.toml]: I don't think we can merge this until we publish a version of these to crates.io and put version numbers here

Comment (ddd-mtl) [reply net/Cargo.toml]: yep. No worries. CI is red all over.

File: net/src/p2p_network.rs [hunk 2]
```diff
@@ -56,6 +57,13 @@ impl P2pNetwork {
                 })
             }
             // Create an InMemoryWorker
```
Comment (zippy) [inline net/src/p2p_network.rs]: Fix comment?

Comment (ddd-mtl) [reply net/src/p2p_network.rs]: yep

**Comment by ddd-mtl** [comment]
@zippy @neonphog @lucksus 
CI is green. Can you please approve?

**Pull request merged**